MFVNeutralino/plugins/Vertexer.cc

```diff
@@ -828,11 +1053,33 @@ void MFVVertexer::produce(edm::Event& event, const edm::EventSetup& setup) {
     }
   }
 
+  if (verbose){
+    for (v[0] = vertices->begin(); v[0] != vertices->end(); ++v[0]) {
+      const track_vec tks = vertex_track_vec(*v[0]);
+      const size_t ntks = tks.size();
+      if (ntks < 3)
+        continue;
+      printf("doing refit on vertex at %7.4f %7.4f %7.4f with %lu tracks\n", v[0]->x(), v[0]->y(), v[0]->z(), ntks);
+      for (size_t i = 0; i < ntks; ++i){
+        printf("  refit %lu will drop tk  pt %7.4f +- %7.4f eta %7.4f +- %7.4f phi %7.4f +- %7.4f dxy %7.4f +- %7.4f dz %7.4f +- %7.4f\n", i, tks[i]->pt(), tks[i]->ptError(), tks[i]->eta(), tks[i]->etaError(), tks[i]->phi(), tks[i]->phiError(), tks[i]->dxy(), tks[i]->dxyError(), tks[i]->dz(), tks[i]->dzError());
+        std::vector<reco::TransientTrack> ttks(ntks-1);
+        for (size_t j = 0; j < ntks; ++j)
+          if (j != i)
+            ttks[j-(j>=i)] = tt_builder->build(tks[j]);
+
+        reco::Vertex vnm1(TransientVertex(kv_reco->vertex(ttks)));
+        printf("    refitted vertex chi2/dof %7.4f with track ", vnm1.normalizedChi2());
+        print_track_set(vnm1);
+      }
+    }
+  }
+
```

Comment:
Could you add a comment for this procedure too? I guess you were testing if the chi2 improved if a track was dropped?

Reply:
This part was added when I found that there are many "good" quality SVs before the nm1 refit and drop process. So I made this part to print out how those tracks are dropped during nm1 process and how a "good" vertex becomes vertices with only 2 tracks in the end.